### ISSUE TITLE
[Feature] Ability to save idents in a shared collection from the browser

### ISSUE THREAD

**Issue by felicienfrancois** [bitwarden/browser#236]
That would makes the Enterprise / Team offer more relevant.

**Comment by ohthehugemanatee** [comment]
I started trying to do this today... but it relies on services for Organizations and Collections. My PR is quickly becoming a monolith, which is definitely a bad idea for a first time patch submission, in a framework and language that's not my primary.

Can we break this into smaller components? It's not _terrible_ to have simple, read-only Organizations and Collections support first... but maybe it would be better to work on support for "manage organizations" and a "manage collections" first. Three small issues is better than one big one...

In case it helps, messy half-written WIP at https://github.com/ohthehugemanatee/browser/tree/share-button

**Comment by kspearrin** [comment]
@ohthehugemanatee Yes, there are a lot of moving parts that have to be completed before this is possible. There is no organization information synced down to the clients at this time.

**Comment by ohthehugemanatee** [comment]
I see that. On the other hand, orgs and collections are simple objects. We don't necessarily need setters for them in the extension yet.  That's how I approached it anyway, and collections were fine. But organizations... it really makes sense for them to be attached to the user's profile (like in web). I felt uncomfortable messing around with that code, though.

I'll keep hacking away at this as I have time, but I will definitely need help. Is there an established process for work-in-progress PRs?

**Comment by kspearrin** [comment]
We would likely want to alter some things on the server first. For example, adding organizations and collections to the sync APIs. You can always open a WIP PR if you like, even if only to convey an idea that we later implement ourselves instead of accepting your PR.

**Comment by ohthehugemanatee** [comment]
for now I've tried to stick with the architecture and patterns in the rest of the codebase, so there's not a lot of implementation information to express. I won't venture into single-handedly submitting a server PR *and* a related browser PR. I'd be happier to have a separate call anyway. So it seems like yeah, I should just put it all into one big PR. Thanks for your help and fast response!
